include/pagmo/algorithms/nsga2.hpp

```diff
@@ -0,0 +1,571 @@
+     *   6            260      0.0018169      0.0873995       0.509662
+     *   7            312     0.00154273      0.0873995       0.492973
+     *   8            364     0.00154273      0.0873995       0.471251
+     *   9            416    0.000379582      0.0873995       0.471251
+     *  10            468    0.000336743      0.0855247       0.432144
+     * @endcode
+     * Gen, is the generation number, Fevals the number of function evaluation used. The ideal point of the current
+     * population follows cropped to its 5th component.
+     *
+     * @param level verbosity level
+     */
+    void set_verbosity(unsigned int level)
+    {
+        m_verbosity = level;
+    };
+    /// Gets the verbosity level
+    /**
+     * @return the verbosity level
+     */
+    unsigned int get_verbosity() const
+    {
+        return m_verbosity;
+    }
+    /// Algorithm name
+    /**
+     * Returns the name of the algorithm.
+     *
+     * @return <tt> std::string </tt> containing the algorithm name
+     */
+    std::string get_name() const
+    {
+        return "NSGA-II";
+    }
+    /// Extra informations
+    /**
+     * Returns extra information on the algorithm.
+     *
+     * @return an <tt> std::string </tt> containing extra informations on the algorithm
+     */
+    std::string get_extra_info() const
+    {
+        std::ostringstream ss;
+        stream(ss, "\tGenerations: ", m_gen);
+        stream(ss, "\n\tCrossover probability: ", m_cr);
+        stream(ss, "\n\tDistribution index for crossover: ", m_eta_c);
+        stream(ss, "\n\\tMutation probability: ", m_m);
+        stream(ss, "\n\tDistribution index for mutation: ", m_eta_m);
+        stream(ss, "\n\tSize of the integer part: ", m_int_dim);
+        stream(ss, "\n\tSeed: ", m_seed);
+        stream(ss, "\n\tVerbosity: ", m_verbosity);
+        return ss.str();
+    }
+    /// Get log
+    /**
+     * A log containing relevant quantities monitoring the last call to evolve. Each element of the returned
+     * <tt> std::vector </tt> is a nsga2::log_line_type containing: Gen, Fevals, ideal_point
+     * as described in nsga2::set_verbosity
+     * @return an <tt> std::vector </tt> of nsga2::log_line_type containing the logged values Gen, Fevals,
+     * ideal_point
+     */
+    const log_type &get_log() const
+    {
+        return m_log;
+    }
+    /// Object serialization
+    /**
+     * This method will save/load \p this into the archive \p ar.
+     *
+     * @param ar target archive.
+     *
+     * @throws unspecified any exception thrown by the serialization of the UDP and of primitive types.
+     */
+    template <typename Archive>
+    void serialize(Archive &ar)
+    {
+        ar(m_gen, m_cr, m_eta_c, m_m, m_eta_m, m_e, m_int_dim, m_seed, m_verbosity, m_log);
+    }
+
+private:
+    vector_double::size_type tournament_selection(vector_double::size_type idx1, vector_double::size_type idx2,
```

Comment:
If these helper methods do not require access to data in ``this``, you could just make them ``static``.